### PR TITLE
Fixes codeblock indentation in order to make them actually work

### PR DESCRIPTION
**Pull request by arielf212**
Currently the documentation website shows broken code blocks, this PR provides a fix to that :)

### CODE REVIEW

**Review by vadorovsky — CHANGES_REQUESTED** [review]
Thanks a lot for the fix! :slightly_smiling_face: 

Unfortunately, markdownlint complains about that:

```
Error: docs/book/programs/cgroup-skb.md:43 MD046/code-block-style Code block style [Expected: indented; Actual: fenced] https://github.com/DavidAnson/markdownlint/blob/v0.36.1/doc/md046.md
```

But honestly, I think this error is 🐂 :hankey:, since the "indented" code blocks simply don't work.

So I think we could just try to ignore that error. According to:

https://github.com/DavidAnson/markdownlint-cli2-action?tab=readme-ov-file#config-optional
https://github.com/DavidAnson/markdownlint-cli2

We could try changing the markdownlint job definition in `.github/workflows/ci.yml` to:

```yaml
      - uses: DavidAnson/markdownlint-cli2-action@v18
        with:
          config: .markdownlint.json
          globs: '**/*.md'
```

And then add the `.markdownlint.json` file with the following content:

```json
{
  "code-block-style": {
    "style": "fenced"
  }
}
```

I think that should fix the CI failure.

**Comment by arielf212** [comment]
Done!
Thank you for your guidance! :)

**Review by vadorovsky — CHANGES_REQUESTED** [review]
Thanks!

CI is now rightfuly pointing out some places where code blocks are indented. Can you remove the indentation there as well?

**Comment by arielf212** [comment]
Yeah, just hopped in to see what the CI is yapping about now.
Seems like I can't fix most of those problems, since that would break the intended look of the page.
In general, I am not sure this utility is very useful for mkdocs, since mkdocs uses a lot of non-standard markdown extensions :)

**Comment by arielf212** [comment]
I found myself copy-pasting a lot of enable/disable pragmas into the codebase, which I found really ugly.
As such, I just disabled two very noisy rules and fixed the remaining one.
As I described in the commit message, the codeblock rule fundamentally clashes with how mkdocs works,
while the second rule (dollar sign) clashes with the already-set convention for this repo.

File: .markdownlint.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+    "code-block-style": false,
+    "commands-show-output": false
```
Comment (tamird) [inline .markdownlint.json:3]: i think this is not correct. see the `aya-tool` example which does show its output. Just remove the `$` signs where the linter complains.

Comment (arielf212) [reply .markdownlint.json:3]: This does not stop anyone from using the dollar sign in order to show the output.
This is just the linter config - it disables the lint which tells you to remove the dollar sign in unneeded places, which seems to be a convention in this repo.

Comment (vadorovsky) [reply .markdownlint.json:3]: I agree with disabling this lint, it makes no sense to me (enforcing the lack of `$`) and the name is misleading.

I would say "please add the comment", but JSON files don't support comments, FML...

**Comment by vadorovsky** [comment]
>In general, I am not sure this utility is very useful for mkdocs, since mkdocs uses a lot of non-standard markdown extensions :)

To be honest, the only reason I decided to introduce it is that we used to have a lot of places with over 200 charaters, weird whitespaces and inconsistent heading. And also a lot of PR introducing more of these. :sweat_smile:  I agree that many  markdownlint rules are questionable and not compatible with mkdocs. Unfortunately, I don't think there is a good alternative. So I think we just need to figure out a minimal set of rules which make sense for us to use.

File: .markdownlint.json (hunk 1 above)
Comment (tamird) [reply .markdownlint.json:3]: It does make some sense, the rationale is given: https://github.com/updownpress/markdown-lint/blob/master/rules/014-commands-show-output.md

Comment (arielf212) [reply .markdownlint.json]: Your repo your rules :P
Pushed a fix.

**Review by tamird — APPROVED** [review]
LGTM. Mind squashing these down to a single commit with a sensible message?

File: .markdownlint.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "code-block-style": false
```
Comment (tamird) [inline .markdownlint.json:2]: oh wait, can we make this fenced? or why not?

Comment (arielf212) [reply .markdownlint.json:2]: Having this be `fenced` messes with places which have to be indented, like inside an admonition

Comment (arielf212) [reply .markdownlint.json:2]: also, squashed

Comment (tamird) [reply .markdownlint.json:2]: There's some advice in https://github.com/DavidAnson/markdownlint/issues/207 about how to support admonitions. Any interest in making that work in this PR?

**Review by tamird — APPROVED** [review]
Alright, we can tackle the admonition thing in another PR.

**Comment by arielf212** [comment]
@tamird Seems like the "solution" provided in the link you linked isn't actually relevant to our situation since you can't load plugins in the cli.
As such, there is no easy fix for this as far as I see :)

**Comment by tamird** [comment]
Could you resolve the conflict?

**Review by tamird — APPROVED** [review]

**Pull request merged**